packages/fxa-settings/src/models/Account.ts

```diff
@@ -585,17 +586,40 @@ export class Account implements AccountData {
   }
 
   async changePassword(oldPassword: string, newPassword: string) {
-    const response = await this.withLoadingStatus(
-      this.authClient.passwordChange(
-        this.primaryEmail.email,
-        oldPassword,
-        newPassword,
-        sessionToken()!,
-        {
-          keys: true,
-        }
-      )
-    );
+    const currentSessionToken = sessionToken()!;
+
+    // Check if we have a JWT for MFA-protected password changes
+    const hasMfaJwt = JwtTokenCache.hasToken(currentSessionToken, 'password');
+
+    let response;
+    if (hasMfaJwt) {
```

Comment:
If the jwt is missing. I think we should error out here. This state is now considered invalid.